### ISSUE TITLE
ribbonplot goes beyond particle bounds

### ISSUE THREAD

**Issue by cscherrer** [baggepinnen/MonteCarloMeasurements.jl#26]
What makes MonteCarloMeasurements so apealling to me is the ability to approximate distributions that are very different from Gaussian. But some plotting methods still seem to approximate the approximation by putting things in terms of a Gaussian. 

In particular, I'm thinking of
```julia
help?> ribbonplot
search: ribbonplot ribbonplot!

  ribbonplot(x,y,[q=2])

  Plots a vector of particles with a ribbon representing q*std(y). Default width is
  2σ
```

The 2σ approximation is especially problematic in cases with bounded support, where the plot often crosses into zero-probability land. The 2σ width is presumably to be a stand-in for a credible interval. But we already have something much more precise, so why not use it? 

To me it would be much more natural to have two methods:

- `ribbonplot(x,y,[q::Real])` could give a ribbon spanning from the `(1-q)/2` quantile to the `(1+q)/2` quantile (so the coverage area is `q`)
- `ribbonplot(x,y,[q::(Real,Real)])` could give a ribbon spanning from the `q[1]` quantile to `q[2]`.

Do you see any drawbacks to this approach for your use cases?

**Comment by baggepinnen** [comment]
I'm inclined to agree with you, I have also found the ribbonplot to be problematic. In the meantime, I think `errorbarplot` is close to what you are looking for, although visually different.

**Comment by cscherrer** [comment]
Thanks. This one seems to always use a centered interval of probability `max(q,1-q)`, is that right?

I really need to get around to learning the Plots recipe stuff, implementations are not yet really clear to me. I was thinking it would be great to have an `eachplot` that just does combines the plots per-particle (with some alpha). I tend to use those a lot, like [this](https://cscherrer.github.io/post/max-profit/plots/max_of_means.png). Any resources you've found useful?

**Comment by baggepinnen** [comment]
Alright, I changed `ribbonplot` to behave the same way as `errorbarplot`, i.e., to use quantiles instead of stds. I also added the option to supply both upper and lower quantiles to both of these plots. Is this roughly what you had in mind? 

If you write some custom plot recipes you think might come in handy for others as well, feel free to subit a PR. Getting a hang of how to write recipes took me a while, but now I can do it mostly without referring to the manual. The biggest drawback I see with the `@userplot` recipes is that you have to fall back to python style checking of input arguments and can not use multiple dispatch, at least not without some clever function barriers. The plots manual on recipes is the only reference I've ever used (and contributed to as note-to-self). It's still not that easy to follow, but gets you there if the need is strong enough ;)

Btw, thanks for the mention in your juliacon talk, I just saw it!

**Comment by baggepinnen** [comment]
Note to self: need to fix this function as well
https://github.com/baggepinnen/MonteCarloMeasurements.jl/blob/5c4d1383405fe3cc013eb4a54d35071e4d013460/src/plotting.jl#L88

**Comment by baggepinnen** [comment]
This issue should be resolved completely after 945e34f005e8493ced676292578026abedb2816c
